media-gfx/photoqt/photoqt-4.7.ebuild

```diff
@@ -0,0 +1,98 @@
+# Copyright 1999-2025 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{10..12} )
+
+ECM_KDEINSTALLDIRS=false
+KFMIN=6.5.0
+inherit ecm optfeature python-single-r1 toolchain-funcs
+
+DESCRIPTION="Simple but powerful Qt-based image viewer"
+HOMEPAGE="https://photoqt.org/"
+SRC_URI="https://photoqt.org/downloads/source/${P}.tar.gz"
+
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="barcode chromecast devil exif freeimage geolocation graphicsmagick +imagemagick lcms mpv pdf raw vips"
+REQUIRED_USE="
+	?? ( imagemagick graphicsmagick )
+	chromecast? ( ${PYTHON_REQUIRED_USE} )
+"
+
+COMMON_DEPEND="
+	app-arch/libarchive:=
+	dev-libs/pugixml
+	dev-qt/qtbase:6[dbus,concurrent,gui,network,sql,widgets,xml]
+	dev-qt/qtdeclarative:6
+	dev-qt/qtimageformats:6
+	dev-qt/qtmultimedia:6[qml]
+	dev-qt/qtsvg:6
+	barcode? ( media-libs/zxing-cpp:= )
+	chromecast? (
+		${PYTHON_DEPS}
+		$(python_gen_cond_dep 'dev-python/pychromecast')
+	)
+	devil? ( media-libs/devil )
+	exif? ( media-gfx/exiv2:=[bmff] )
+	freeimage? ( media-libs/freeimage )
+	graphicsmagick? ( media-gfx/graphicsmagick:=[cxx] )
+	imagemagick? ( media-gfx/imagemagick:=[cxx] )
+	lcms? ( media-libs/lcms:2 )
+	mpv? ( media-video/mpv:=[libmpv] )
+	pdf? ( app-text/poppler[qt6] )
+	raw? ( media-libs/libraw:= )
+	vips? (
+		dev-libs/glib:2
+		media-libs/vips:=
+	)
+"
+RDEPEND="
+	${COMMON_DEPEND}
+	geolocation? (
+		dev-qt/qtlocation:6
+		dev-qt/qtpositioning:6[qml]
+	)
+"
+DEPEND="
+	${COMMON_DEPEND}
+	vips? ( x11-base/xorg-proto )
+"
+BDEPEND="
+	dev-qt/qttools:6[linguist]
+	virtual/pkgconfig
+	chromecast? ( ${PYTHON_DEPS} )
```

Comment:
If the Python deps are only needed conditionally, we should define a `pkg_setup` where we conditionally call `python-single-r1_pkg_setup` for `use chromecast`.